request_errors.go

```diff
@@ -15,7 +15,7 @@ func (r RequestError) Error() string {
 	return fmt.Sprintf("An error occurred while calling %s", r.Url)
 }
 
-func requestErrorFromOauthError(err error) error {
+func RequestErrorFromOauthError(err error) error {
```

Comment:
I'm not sure this needs to be exported. I wonder if https://golang.org/doc/go1.13#error_wrapping will assist us in providing the desired experience?